### PR TITLE
Feature/list users group endpoint

### PR DESCRIPTION
**Pull request by HeJones**
### What

Add endpoint 
GET /v1/groups/{id}/members
to list the users belonging to a group

### How to review

Ensure unit tests and component tests pass

Run `make debug`:
when you make a GET request to the following endpoint: http://localhost:25600/v1/groups/role-admins/members you should see 1 users
when you make a GET request to the following endpoint: http://localhost:25600/v1/groups/test-group-0/members you should see 178 users (there is a count variable at the bottom of the list)
### Who can review

Anyone

### CODE REVIEW

File: api/groups.go [hunk 1]
```diff
@@ -67,6 +69,59 @@ func (api *API) AddUserToGroupHandler(ctx context.Context, w http.ResponseWriter
...
+		firstTimeCheck = true
+
+		groupMembersRequest := group.BuildListUsersInGroupRequestWithNextToken(api.UserPoolId, nextToken)
+		groupMembersResponse, err := api.CognitoClient.ListUsersInGroup(groupMembersRequest)
+		if err != nil {
+			return nil, err
+		}
+
+		listOfUsers = append(listOfUsers, groupMembersResponse.Users...)
+		if groupMembersResponse.NextToken != nil {
```
Comment (daiLlew) [inline api/groups.go:117]: The `else` isn't required - IMO less indentation is better/easier to read.
```
nextToken = ""
if groupMembersResponse.NextToken != nil {
    nextToken = *groupMembersResponse.NextToken
} 
```
This is personal opinion and I'm not requiring you to make this change - just offering my opinion.

File: models/groups.go [hunk 2]
```diff
@@ -96,6 +97,22 @@ func (g *Group) BuildListUsersInGroupRequest(userPoolId string) *cognitoidentity
...
+			GroupName:  &g.Name,
+			UserPoolId: &userPoolId,
+		}
+	} else {
+		return &cognitoidentityprovider.ListUsersInGroupInput{
+			GroupName:  &g.Name,
+			UserPoolId: &userPoolId,
+			NextToken:  &nextToken,
+		}
+	}
```
Comment (daiLlew) [inline models/groups.go]: The `else` not required here as there is a return inside the `if` block. Again no functional difference its just less code/indentation which I think makes it easier to read.
```
if nextToken == "" {
    return &cognitoidentityprovider.ListUsersInGroupInput{
	GroupName:  &g.Name,
	UserPoolId: &userPoolId,
}

return &cognitoidentityprovider.ListUsersInGroupInput{
    GroupName:  &g.Name,
    UserPoolId: &userPoolId,
    NextToken:  &nextToken,
}
```

File: models/groups_test.go [hunk 3]
```diff
@@ -249,3 +250,34 @@ func TestGroup_BuildSuccessfulJsonResponse(t *testing.T) {
...
+			Name: "role-test",
+		}
+		userPoolId := "euwest-99-aabbcc"
+		nextToken := ""
+
+		response := group.BuildListUsersInGroupRequestWithNextToken(userPoolId, nextToken)
+
+		So(reflect.TypeOf(*response), ShouldEqual, reflect.TypeOf(cognitoidentityprovider.ListUsersInGroupInput{}))
+		So(*response.UserPoolId, ShouldEqual, userPoolId)
+		So(*response.GroupName, ShouldEqual, group.Name)
```
Comment (daiLlew) [inline models/groups_test.go:266]: Add an assertion to verify that `NextToken` is nil/empty

Comment (HeJones) [reply models/groups_test.go:266]: I did initially try with this ```So(*response.NextToken, ShouldBeNil)``` but I got the following error: ```runtime error: invalid memory address or nil pointer dereference ```. Do you know if there is a way to write a test without getting this error please?

Comment (HeJones) [reply models/groups_test.go:266]: I think it works if I have `So(response.NextToken, ShouldBeNil)`

File: models/groups.go (hunk 2 above)
Comment (HeJones) [reply models/groups.go]: Thank you

File: api/groups.go (hunk 1 above)
Comment (HeJones) [reply api/groups.go:117]: Thank you

**Review by daiLlew — APPROVED** [review]

**Pull request merged**